File: app/src/main/java/com/lingoswap/domain/usecase/auth/LoginUseCase.java
package com.lingoswap.domain.usecase.auth;

import com.lingoswap.data.model.AuthResponse;
import com.lingoswap.data.repository.RepositoryCallback;
import com.lingoswap.domain.repository.AuthRepository;

import javax.inject.Inject;

public class LoginUseCase {
    private final AuthRepository repository;

    @Inject
    public LoginUseCase(AuthRepository repository) {
        this.repository = repository;
    }

    public void execute(String email, String password, RepositoryCallback<AuthResponse> callback) {
        repository.login(email, password, callback);
    }
}

File: app/src/main/java/com/lingoswap/domain/usecase/auth/LoginUseCase.kt
package com.lingoswap.domain.usecase.auth

import com.lingoswap.data.model.request.LoginRequest
import com.lingoswap.data.model.response.AuthResponse
import com.lingoswap.domain.repository.AuthRepository
import com.lingoswap.utils.Resource
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class LoginUseCase @Inject constructor(
    private val repository: AuthRepository
) {
    suspend operator fun invoke(request: LoginRequest): Flow<Resource<AuthResponse>> {
        return repository.login(request)
    }
}
